src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -151,16 +152,35 @@ public int CurrentWorkspaceIndex
         {
             get
             {
+                // It is safe to assume that DynamoModel.CurrentWorkspace is
+                // update-to-date.
                 var viewModel = workspaces.FirstOrDefault(vm => vm.Model == model.CurrentWorkspace);
                 var index = workspaces.IndexOf(viewModel);
+
+                // As the getter could aslo be triggered by the change of model,
+                // we need to update currentWorkspaceViewModel here. 
+                if (currentWorkspaceViewModel != viewModel)
+                    currentWorkspaceViewModel = viewModel;
+
                 return index;
             }
             set
```

Comment:
Not all switching workspace going through ``SwitchTabCommand``. One is adding home workspace model, which 1) deletes current home workspace model, and consequently delete current home workspace view model (and cause ``CurrentWorkspaceIndex`` is triggered as workspace view mode list is the data source of DynamoView 2) and add new home workspace and new home workspace view model. During this periods, at some points ``CurrentWorkspace`` hasn't be properly set or is in invalid state.